tuf/ngclient/updater.py

```diff
@@ -471,7 +503,67 @@ def _preorder_depth_first_walk(
         # If this point is reached then target is not found, return None
         return None
 
+    def _download_directory(self, folder: str) -> None:
+        # Downloading all metadata inside the specified folder
+
+        path = (
+            f"{BASE_URL}/repository_data/TAP 14"  # File path to download from
+        )
+        path = os.path.join(path, folder)  # Add the folder to the path
+
+        # Repository might be storing data in different formats.
+
+        files = next(os.walk(path))[2]  # Get all the files inside the list
+        for file in files:
+            if file.endswith(".json"):
+                self._download_metadata(file)
+
 
 def _ensure_trailing_slash(url: str) -> str:
     """Return url guaranteed to end in a slash"""
     return url if url.endswith("/") else f"{url}/"
+
+
+def _get_spec_version(
+    repository_versions: List[str],
+    spec_version: str,
+    supported_versions: List[str],
+) -> str:
```

Comment:
Oh, we need to return the actual spec version too!

Maybe make the return type `Tuple[Optional[str], Optional[str]]`.

Reply:
How about `Tuple[[str], Optional[str]]` ? `spec_version` is supposed to be either `""` or `[versions, ...]` whereas `warning` can be `None`.

Reply:
I see `return (str(spec_version), warning)` below, which suggests:

`Tuple[str, Optional[str]]`.

Reply:
Ah, I meant that, oops. Will add that then.